### PR TITLE
Add Support for Fanju 3378 temperature sensor

### PR DESCRIPTION
**Pull request by Blokkendoos**
4) Add a page to the pilight manual

### CODE REVIEW

File: docs/source/protocols/433.92/weather/fanju.rst [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+| show-temperature     | 1           | 1 or 0     | '1' display the temperature value                         |
++----------------------+-------------+------------+-----------------------------------------------------------+
+| show-humidity        | 1           | 1 or 0     | '1' display the humidity value                            |
++----------------------+-------------+------------+-----------------------------------------------------------+
+| show-battery         | 1           | 1 or 0     | '1' display the battery value                             |
++----------------------+-------------+------------+-----------------------------------------------------------+
+
+.. rubric:: Comments
+
+This weather station was available at AliExpress. Description for the device could be: “Fanju Wireless Digital Temperature Humidity Wireless Sensor meter In/Outdoor used with Weather Station”, and “Match 3378/74/79C/65/89 Indoor Receiver”. The sensor can be identified by the option to choose from three different channels.
```
Comment (CurlyMoo) [inline docs/source/protocols/433.92/weather/fanju.rst]: `Was` available of `is`?

**Comment by CurlyMoo** [comment]
Where are the actual protocol C file gone?

**Comment by Blokkendoos** [comment]
The protocol code is in the 'rewrite' branch. The documentation is in 'staging'.
Or maybe you mean something else?

**Comment by CurlyMoo** [comment]
> Or maybe you mean something else?

The code from staging should be there as well, because it has a slighty different syntax. Protocol in rewrite don't automatically end up in staging. Having the unittests is just for future purposes, would i change the logic of the protocols.

**Comment by Blokkendoos** [comment]
Hello CurlyMoo,

thanks for your explanation. As you can see (in GitHub), the protocol code has been added to ‘staging’.
Should there be still something missing or should you have any remarks, please let me know.

Regards,
Johan

File: libs/pilight/protocols/433.92/fanju.c [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+			mask ^= 0x9;
+		}
+		if(binary_cpy[i] == 1) {
+			checksum_calc ^= mask;
+		}
+	}
+	if(checksum != checksum_calc) {
+		logprintf(LOG_ERR, "fanju: parsecode - invalid checksum: %d  calc: %d", checksum, checksum_calc);
+		return;
+    }
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/fanju.c]: This accolade is not properly alligned.

File: libs/pilight/protocols/433.92/fanju.c [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+		while(tmp) {
+			if(fabs(tmp->id - id) < EPSILON) {
+				match = 1;
+				break;
+			}
+			tmp = tmp->next;
+		}
+
+		if(match == 0) {
+			if((snode = MALLOC(sizeof(struct settings_t))) == NULL) {
```
Comment (CurlyMoo) [inline libs/pilight/protocols/433.92/fanju.c:208]: Alignment

Comment (Blokkendoos) [reply libs/pilight/protocols/433.92/fanju.c:208]: I'm so sorry, has been fixed now (and I should not be mixing Vim and TextWrangler)

**Pull request merged**